cmd/celestia/main.go

```diff
@@ -35,6 +38,13 @@ func main() {
 }
 
 func run() error {
+	seed := make([]byte, 32)
+	_, err := crypto_rand.Read(seed)
+	if err != nil {
+		return err
+	}
+	math_rand.Seed(int64(binary.LittleEndian.Uint64(seed)))
```

Comment:
This definitely works to make picking the quadrant less deterministic. That said, I feel the code should be closer to where the randomness is actually used. Maybe in an init. Also, is there a way to use your own instantiation of the a prng instead of using the global one from math/rand? Then it could be instantiated as the above without having any side-effects.

Reply:
We could `init` it somewhere and then pass around a `*rand.Rand` maybe?

Reply:
> We could init it somewhere and then pass around a *rand.Rand maybe?

Yeah, or even construct that rand.Rand where needed once as an unexported field. Both works.

Reply:
* There is global randomness and there could be other places that use the global randomness, not only quadrant requesting logic. It is ok to set up this global randomness on the application startup. It is normal in Go to use global randmoness for cases where cryptographic randmoness is not required.
* Whether quadrant fetching logic should use or not use global randomness is orthogonal.

Reply:
Regarding the change itself. Seeding with `time.Now().Unix()` is enough and feeding math_rand with crypto_rand's number does not give cryptographic secureness. 
> Funnily, I had similar impression, but than Go's crypto lead explained me this, somewhere in celestia-core repo. @liamsi should remember that PR. The lead even made a change to a GO's doc pointing this out and linked the lazyledger-core back then into the commit history.

Reply:
Wanna make an issue with some justification about using crypto.Rand instead? 
> Every time we discuss this, I wonder if we should not simply use crypto/rand everywhere instead :D

Reply:
Yeah, can do: https://github.com/celestiaorg/celestia-node/issues/865